arrow-libs/core/arrow-core/src/commonMain/kotlin/arrow/core/NonEmptySet.kt

```diff
@@ -0,0 +1,34 @@
+package arrow.core
+
+public class NonEmptySet<out T>(
+  private val first: T,
+  private val rest: Set<T>,
+) : AbstractSet<T>() {
+
+  public val elements: Set<T> by lazy { setOf(first) + rest }
```

Comment:
Does the benefit of deferred initialisation here outweigh the potentially significant cost of using a lazy property?